website/project/model.py

```diff
@@ -852,16 +856,48 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
     # Primary institution node is attached to
     primary_institution = fields.ForeignField('institution')
 
+    affiliated_institutions = fields.ForeignField('institution', list=True)
+
     def add_primary_institution(self, user, inst):
-        if inst.auth(user):
-            self.primary_institution = inst
-            self.save()
-            return True
-        return False
+        if not inst.auth(user):
+            raise UserNotAffiliatedError('User is not affiliared with {}'.format(inst.name))
+        if inst == self.primary_institution:
+            return False
+        self.primary_institution = inst
+        if inst not in self.affiliated_institutions:
+            self.affiliated_institutions.append(inst)
+        self.add_log(
+            action=NodeLog.INSTITUTION_ADDED,
```

Comment:
yep, for all these comments, this was meant for primary institutions only, since we are not currently actually supporting multiple institutions. Instead of changing whether or not add_log gets called, I will change the action name.